mlir/lib/Quantum/Transforms/remove_chained_self_inverse.cpp

```diff
@@ -63,29 +63,11 @@ struct RemoveChainedSelfInversePass
         }
 
         Operation *module = getOperation();
-        Operation *targetfunc;
-
-        WalkResult result = module->walk([&](func::FuncOp op) {
-            StringRef funcName = op.getSymName();
-
-            if (funcName != FuncNameOpt) {
-                // not the function to run the pass on, visit the next function
-                return WalkResult::advance();
-            }
-            targetfunc = op;
-            return WalkResult::interrupt();
-        });
-
-        if (!result.wasInterrupted()) {
-            // Never met a target function
-            // Do nothing and exit!
-            return;
-        }
 
         RewritePatternSet patterns(&getContext());
         populateSelfInversePatterns(patterns);
 
-        if (failed(applyPatternsAndFoldGreedily(targetfunc, std::move(patterns)))) {
+        if (failed(applyPatternsAndFoldGreedily(module, std::move(patterns)))) {
```

Comment:
Is this module the overall big module, or the module for the qnode?

Or, I guess a better question is, do we have control over the module targeted by `runOnOperation`, or do we just trust that it fetches the qnode module?

Reply:
Ah, I just realized it doesn't matter as the applicator visits the nested modules anyway.

Reply:
The frontend will produce code such that the module here will only refer to the nested module. But in principle it could be applied to the root module as well.